lms/templates/courseware/gradebook.html

```diff
@@ -102,39 +104,57 @@
         </%def>
 
-        <tbody>
-          %for student in students:
+       <%
+        student_records_row = ""
+        for student in students:
+          student_records_col = ""
+          for section in student['grade_summary']['section_breakdown']:
+              student_records_col += "{col}".format(col=percent_data( section['percent'], section['detail']))
+          student_records_col += "{col}".format(col= percent_data( student['grade_summary']['percent'], _('Total')))
+
+          student_records_row += "<tr>{cols}</tr>".format(cols=student_records_col)
+       %>
+
+        <tbody id="grade-book-student-records">
           <tr>
-            %for section in student['grade_summary']['section_breakdown']:
-              ${percent_data( section['percent'], section['detail'] )}
-            %endfor
-            ${percent_data( student['grade_summary']['percent'], _('Total'))}
+            <td>
+              <i class="icon fa fa-spinner fa-pulse fa-5x"></i>
+            </td>
           </tr>
-          %endfor
         </tbody>
       </table>
-    </div>
-    <span class="grade-book-footer">
-        %if page["previous_offset"] is not None:
-          <a href="${page_url}?offset=${page['previous_offset']}"
-             class="sequence-nav-button button-previous">
-            <span class="icon fa fa-chevron-left" aria-hidden="true"></span><span class="sr">${_('previous page')}</span>
-          </a>
+      <script>
+        %if student_records_row is not "":
+          $("#grade-book-student-records").html("${student_records_row}");
         %endif
-
-        ${_('Page')} ${page["page_num"]} ${_('of')} ${page["total_pages"]}
-
-        %if page["next_offset"] is not None:
-          <a href="${page_url}?offset=${page['next_offset']}"
-             class="sequence-nav-button button-next">
-            <span class="icon fa fa-chevron-right" aria-hidden="true"></span><span class="sr">${_('next page')}</span>
-          </a>
-        %endif
-    </span>
+      </script>
+    </div>
+    %if not ccx:
+      <span class="grade-book-footer">
+          %if page["previous_offset"] is not None:
+            <a href="${page_url}?offset=${page['previous_offset']}"
+               class="sequence-nav-button button-previous">
+              <span class="icon fa fa-chevron-left" aria-hidden="true"></span><span class="sr">${_('previous page')}</span>
+            </a>
+          %endif
+
+          ## Translators: Current page number out of total pages in grade book.
+          ${_('Page {page_num} of {total_pages}').format(page_num=page["page_num"], total_pages=page["total_pages"])}
+
+          %if page["next_offset"] is not None:
+            <a href="${page_url}?offset=${page['next_offset']}"
+               class="sequence-nav-button button-next">
+              <span class="icon fa fa-chevron-right" aria-hidden="true"></span><span class="sr">${_('next page')}</span>
+            </a>
+          %endif
+      </span>
+    %endif
```

Comment:
@pdpinch this was a mistake. I refactored and fixed issue, please see new commit.